packages/now-next/src/index.ts

```diff
@@ -913,25 +913,36 @@ export const build = async ({
 
       debug(`node-file-trace result for pages: ${fileList}`);
 
+      const lstatResults: { [key: string]: ReturnType<typeof lstat> } = {};
+
       const collectTracedFiles = (
         reasons: NodeFileTraceReasons,
         files: { [filePath: string]: FileFsRef }
-      ) => (file: string) => {
+      ) => async (file: string) => {
         const reason = reasons[file];
         if (reason && reason.type === 'initial') {
           // Initial files are manually added to the lambda later
           return;
         }
-        const { mode } = lstatSync(path.join(workPath, file));
+        const filePath = path.join(workPath, file);
+
+        if (!lstatResults[filePath]) {
+          lstatResults[filePath] = lstat(filePath);
```

Comment:
Sema should be here, not in `Promise.all`.

Reply:
Added